katdal/test/test_chunkstore_dict.py

```diff
@@ -0,0 +1,25 @@
+################################################################################
+# Copyright (c) 2017-2018, National Research Foundation (Square Kilometre Array)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Tests for :py:mod:`katdal.chunkstore_dict`."""
+
+from katdal.chunkstore_dict import DictChunkStore
+from katdal.test.test_chunkstore import ChunkStoreTestBase
+
+
+class TestDictChunkStore(ChunkStoreTestBase):
+    def setup(self):
+        self.store = DictChunkStore(x=self.x, y=self.y)
```

Comment:
This is already tested in the basic `ChunkStoreTestBase.test_put_and_get`...

Quoting:

```Python
        # Stored object has fewer bytes than expected (and wrong dtype)
        assert_raises(BadChunk, self.store.get, name, s, self.y.dtype)
```

For chunk stores with no idea of the underlying dtype, using the wrong dtype will cause a data size mismatch and hence `BadChunk`. For the `DictChunkStore` which would have allowed arbitrary slices on the underlying array otherwise, the dtype comparison kicks in and also leads to `BadChunk`.